tests/source/use.rs

```diff
@@ -0,0 +1,95 @@
+use foo :: bar
+;
+
```

Comment:
Can you rename `tests/[source/target]/use.rs` to `tests/[source/target]/issue_5852.rs` so we can associate these test cases with that issue. Also, can you include the code snippets from the issue as test cases too.

Reply:
The code snippets are semantically equivalent to the third and sixth cases in the test file.

Reply:
It never hurts to have more tests 😁. I'd appreciate it if you could add those code snippets.

Reply:
@ytmimi done